packages/labs/src/components/select/multiSelect.tsx

```diff
@@ -0,0 +1,256 @@
+        const { tagInputProps = {}, popoverProps = {} } = this.props;
+        const { handleKeyDown, handleKeyUp, query } = listProps;
+        const defaultInputProps: HTMLInputProps = {
+            placeholder: "Search...",
+            ...this.props.tagInputProps.inputProps,
+            // tslint:disable-next-line:object-literal-sort-keys
+            onChange: this.handleQueryChange,
+            ref: this.refHandlers.input,
+            value: query,
+        };
+
+        this.queryListKeyDown = handleKeyDown;
+
+        return (
+            <Popover
+                autoFocus={false}
+                canEscapeKeyClose={true}
+                enforceFocus={false}
+                isOpen={this.state.isOpen}
+                position={Position.BOTTOM_LEFT}
+                {...popoverProps}
+                className={classNames(listProps.className, popoverProps.className)}
+                onInteraction={this.handlePopoverInteraction}
+                popoverClassName={classNames("pt-multi-select-popover", popoverProps.popoverClassName)}
+                popoverDidOpen={this.handlePopoverDidOpen}
+                popoverWillOpen={this.handlePopoverWillOpen}
+            >
+                <div
+                    onKeyDown={this.handleKeyDown}
+                    onKeyUp={this.state.isOpen && handleKeyUp}
+                >
+                    <TagInput
+                        inputProps={defaultInputProps}
+                        {...tagInputProps}
+                        className={classNames("pt-multi-select", tagInputProps.className)}
+                        values={this.props.selectedItems.map(this.props.tagRenderer)}
+                    />
+                </div>
+                <div onKeyDown={handleKeyDown} onKeyUp={handleKeyUp}>
+                    <ul className={Classes.MENU} ref={listProps.itemsParentRef}>
+                        {this.renderItems(listProps)}
+                    </ul>
+                </div>
+            </Popover>
+        );
+    }
+
+    private renderItems({ activeItem, filteredItems, handleItemSelect }: IQueryListRendererProps<T>) {
+        const { itemRenderer, noResults } = this.props;
+        if (filteredItems.length === 0) {
+            return noResults;
+        }
+        return filteredItems.map((item, index) => itemRenderer({
+            index,
+            item,
+            handleClick: (e) => handleItemSelect(item, e),
+            isActive: item === activeItem,
+        }));
+    }
+
+    private handleQueryChange = (event: React.FormEvent<HTMLInputElement>) => {
+        const query = event.currentTarget.value;
+        this.setState({ query, isOpen: !(query.length === 0 && this.props.openOnKeyDown) });
+    }
+
+    private handleItemSelect = (item: T, event: React.SyntheticEvent<HTMLElement>) => {
+        this.input.focus();
+        // make sure the query is valid by checking if activeItem is defined
+        if (this.state.activeItem) {
```

Comment:
please compare against null (`this.state.activeItem != null`), don't implicitly coerce to boolean